llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -1146,7 +1146,7 @@ const SCEV *ScalarEvolution::getLosslessPtrToIntExpr(const SCEV *Op,
     return S;
 
   if (!getDataLayout().isByteAddressable())
-    return Op;
+    return getCouldNotCompute();
```

Comment:
@alexp-sssup this makes sense to me. But I remember you and Carlo discussing about SCEV, so maybe you have more insight.

Without doing this, there is a vector of pointers that crashes because it is later expected to have been converted to a vector of ints. the conversion happens after the early exit here.